Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -186,7 +188,11 @@ internal HttpMessageHandler SetupCustomHttpHandler(HttpClientConfig config)
                         entry = entry.Replace(".", "[.]");
                         // * -> .*  because * is a quantifier and need a char or group to apply to
                         entry = entry.Replace("*", ".*");
-
+                        
+                        entry = entry.StartsWith("^") ? entry : $"^{entry}";

```

Comment:
In the case of the jdbc driver it has the same behavior, it uses string.matches function so the full string should match with the host value, so if we provide snowflakecomputing.com as the nonproxyhosts it will not be bypassed.